NetID Spawning Issue Fix

slikeytre opened the pull request:
Fixs NetID going missing when spawning Player. Not sure if it causes other problems.

slikeytre commented:
Not sure on how to produce the error but i have a screenshot showing how / why it was happening. https://cdn.discordapp.com/attachments/498669222587531265/532059207130808358/unknown.png

At Char select soon as i would select a char a frame could pass by where net ID had 0 observers and this would lead to it being killed and login sequence would get stuck on char select.

slikeytre commented:
Left side of screenshot is a successful login and right is one that failed

vis2k commented:
can you reproduce that reliably? if yes, can you tell us how?
if not, can you try removing all networkproximitycheckers and then create your own where oncheckobserver returns false and onrebuildobservers returns true, without adding anything to observers?

slikeytre commented:
9/10 logins resulted in the Bug. Spent 3 days trying to fix it. Talked to paul he said remove prox detector and see if bug occurs. I removed it, 10/10 logins 0 issues.  So from there i added debug outputs all over network id to try and see when it was happening and why. Everytime i had a failed login there was 1 Frame where observers count was equal to 0 during the login process. Anytime i had a successful login the only difference was observer count wasn't 0 at all during the login process. I also noticed that visibility was set to false on failed login attempts but when i had a successful login that didn't happen. So i simply added the null check for local connection(assuming this was why observer count was 0).  This fixed the issue 100%. My initial fix was to just disable to prox detector til i was on navmesh but I figured someone else will hit this bug that might not have a navmesh so i dug into the source to find the problem.   The reason i questioned the fix was does having a null check there result in a legit situation where it would prevent prox from doing its job. As far as fixing the problem it works 100%.   

For reproducing my only thought is make a prox detector fire during the login process.  I dunno why i had the issue other than maybe perfect timing?

vis2k commented:
if you remove proxchecker then by default you will have all observers, not none.

try the proxchecker example that I explained above please

slikeytre commented:
Sorry Vis in middle of adding a large system, Don't have time to test but i can tell you if i remove the null check i submitted I'm only able to login on 1/10 attempts if lucky. I dunno what it's doing to cause it but that works.  Spent 3 days on that annoyance can't waste time atm. Sorry bud :(

vis2k commented:
closing this. please reopen when you can provide a project to reproduce it.
